crates/installer/src/engine.rs

```diff
@@ -68,12 +68,16 @@ impl Installer {
 
         // Figure out where we live right now and exclude the rootfs
         let probe = Builder::default().build()?;
-        let root_nodes = if let Ok(device) = probe.get_rootfs_device("/") {
-            let mut nodes = probe.get_device_chain(&device.path).unwrap_or_default();
-            nodes.push(device.path.into());
-            nodes
-        } else {
-            vec![]
+        let root_nodes = match probe.get_rootfs_device("/") {
+            Ok(device) => {
+                let mut nodes = probe.get_device_chain(&device.path).unwrap_or_default();
+                nodes.push(device.path.into());
+                nodes
+            }
+            Err(e) => {
+                log::error!("Failed to determine the device mounted to `/`: {e}");
+                vec![]
+            }
```

Comment:
Unclear why there even is a fallback to `vec![]` here, I think this function should probably return an error when it can't find any install target partitions to offer the user?

Currently, a later piece of the code (a dialoguer method) panics when this fn returns an empty list of boot / system partitions.